autotest/interface/restful/test_restful_interface_func_pytorch.py

```diff
@@ -46,16 +46,13 @@ def test_ignore_eos_streaming(self):
                 temperature=0.01):
             outputList.append(output)
 
-        assert_chat_completions_stream_return(outputList[0], MODEL_NAME, True,
-                                              False)
-        assert_chat_completions_stream_return(outputList[-1], MODEL_NAME,
-                                              False, True)
-        for index in range(1, len(outputList) - 1):
+        assert_chat_completions_stream_return(outputList[-1], MODEL_NAME, True)
+        for index in range(0, len(outputList) - 1):
             assert_chat_completions_stream_return(outputList[index],
                                                   MODEL_NAME)
         assert outputList[-1].get('choices')[0].get(
             'finish_reason') == 'length'
-        assert len(outputList) == 102
+        assert len(outputList) == 101
```

Comment:
why do we use such strict rules? we can just encode the output and check if the number is equal to `max_tokens`

Reply:
It make sense. I can chage according this.

Reply:
There is a endpoint named `encode` in api_server. You may try it directly.

Reply:
![image](https://github.com/InternLM/lmdeploy/assets/145004780/a4eb8fc3-7c41-483f-8395-b150284cd559)

Reply:
> There is an endpoint named `encode` in api_server. You may try it directly.

used encode to get token numbers?

Reply:
I think @AllentDan means `assert max_tokens==api_client.encode(response, add_bos=False)[1]`